src/main/java/model/Wallet.java

```diff
@@ -0,0 +1,60 @@
+package model;
+
+import exception.InvalidMoneyAmountException;
+
+public class Wallet {
+    private int money;
+
+    /**
+     * @param money: integer
+     * @pre money >= 0
+     */
+    public Wallet(int money) {
+        this.money = Math.max(money, 0);
+    }
+
+    public Wallet() {
+        this.money = 100;
+    }
+
+    /**
+     * @param amount: int
+     * @pre amount >= 0
+     */
+    public void put(int amount) throws InvalidMoneyAmountException {
+        if (amount <= 0) {
```

Comment:
method should throw invalidamountexception only when amount is less than zero and not equal to zero
```suggestion
        if (amount < 0) {
```